include/random_walks/gaussian_accelerated_billiard_walk.hpp

```diff
@@ -110,7 +117,11 @@ struct GaussianAcceleratedBilliardWalk
             _L = compute_diameter<GenericPolytope>::template compute<NT>(P);
             computeLcov(E);
             _E = E;
-            _AA.noalias() = (DenseMT)(P.get_mat() * P.get_mat().transpose());
+            if constexpr (std::is_same<AA_type, Eigen::SparseMatrix<NT>>::value) {
+                _AA = (P.get_mat() * P.get_mat().transpose());
+            } else {
+                _AA.noalias() = (DenseMT)(P.get_mat() * P.get_mat().transpose());
```

Comment:
should we explicitly cast it to `DenseMT`?

Reply:
I'm not sure what you mean, for the optimizations I need it to be in colmajor SparseMatrix format

Reply:
the question is: "Where is the problem with `_AA.noalias() = P.get_mat() * P.get_mat().transpose();` ?